src/client/remerge/tracking.ts

```diff
@@ -187,22 +187,34 @@ function sendTrackingEvent({
     const nonTimeFilters = filters.filter(filter => !isTimeFilter(filter));
 
     const timeShift = essence.timeShift;
+    const idToken = parseIdToken();
 
-    track(eventType, {
-      time: viewStartAt,
-      $duration: viewStartAt && secondsSince(viewStartAt),
-      reportDuration: timeFilter,
-      timeShift: timeShift.value && timeShift.getDescription(),
-      filters: nonTimeFilters.map(filter => filter.dimension),
-      filterDetails: nonTimeFilters,
-      splits: splits.map(split => split.dimension),
-      splitsDetails: splits,
-      kpis,
-      visualization,
-      timeToRender,
-      timezone: essence.timezone.toString(),
-      errorMessage
-    }, onPageUnload);
+    track(
+      eventType,
+      {
+        time: viewStartAt,
+        $duration: viewStartAt && secondsSince(viewStartAt),
+        reportDuration: timeFilter,
+        timeShift: timeShift.value && timeShift.getDescription(),
+        filters: nonTimeFilters.map((filter) => filter.dimension),
+        filterDetails: nonTimeFilters,
+        splits: splits.map((split) => split.dimension),
+        splitsDetails: splits,
+        kpis,
+        visualization,
+        timeToRender,
+        timezone: essence.timezone.toString(),
+        errorMessage,
+        ...(idToken && idToken.user.admin
```

Comment:
FYI: Can we already use optional chaining in this project? 🤔

Reply:
It's a good point, I checked and it was added in Typescript 2.1. This project is on 3.3.1 so we should be safe:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html#object-spread-and-rest